sale_order_requested_delivery/models/sale_order.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    requested_delivery_period_start = fields.Datetime(
+        string="Customer Requested Delivery Start Datetime"
```

Comment:
Maybe light a bit the labels
```suggestion
        string="Requested Delivery - Start"
```